pytorch_lightning/profiler/pytorch.py

```diff
@@ -340,11 +341,11 @@ def _total_steps(self) -> int:
         trainer = self._lightning_module.trainer
         if self._schedule.is_training:
             return trainer.num_training_batches
-        if self._schedule._current_action == "validation_step":
+        if self._schedule._current_action.endswith("validation_step"):
             return sum(trainer.num_val_batches) + sum(trainer.num_sanity_val_batches)
-        if self._schedule._current_action == "test_step":
+        if self._schedule._current_action.endswith("test_step"):
             return sum(trainer.num_test_batches)
-        if self._schedule._current_action == "predict_step":
+        if self._schedule._current_action.endswith("predict_step"):
```

Comment:
these changes are required at other places in the same file. Specifically `has_finished`, `_step` and `num_step`. make sure to check if self._current_action is None before using `endswith` there.

Reply:
completely missed this, thanks so much!